### DISCUSSION TITLE
What is Inode?

### DISCUSSION THREAD

**Discussion by joindevopscloud** [daws-76s/.github#1] [Q&A]
Answer and learn the Linux basics through simple questions

What is Inode??

**Comment by mksdevops7** [comment]
_An **Inode** is like a special ID card for each file on your computer. It holds key information about the file, such as where it's stored and when it was created. Think of it as a unique tag that helps the computer keep everything organized._

**Reply by balajibigdatadevops** [reply]
Inode is basically like metadata means it is unix file system, it maintains all files and directories information, how many files and directories owner and group name and permissions etc.. stat file_name command givens complete information,

**Reply by Mynameispa1** [reply]
In a simple note Inode basically store information about the open files by the process such as size of the file, permission of the files etc..

**Comment by knbob** [comment]
In Linux operating system i-node stores information about a file or a directory. Each file or directory associated with an i-node.
When we create a file or directory, i-node is assigned to it.

**Comment by saipdevops555** [comment]
An inode is a file data structure that stores information about any Linux file except its name and data.

Data is stored on your disk in the form of fixed-size blocks. If you save a file that exceeds a standard block, your computer will find the next available segment on which to store the rest of your file. Over time, that can get super confusing.

That’s where inodes come in. While they don’t contain any of the file’s actual data, it stores the file’s metadata, including all the storage blocks on which the file’s data can be found. 

Information contained in an inode:
- File size
- Device on which the file is stored 
- User and group IDs associated with the file
- Permissions needed to access the file
- Creation, read, and write timestamps
- Location of the data (though not the filepath)

Inodes are also independent of filenames. That means you can copy a single file, rename it, and still have it point to the same inode as the original.
